oneview_redfish_toolkit/strategy_multiple_oneview.py

```diff
@@ -0,0 +1,98 @@
+# under the License.
+
+# Modules own libs
+from oneview_redfish_toolkit import multiple_oneview
+
+
+def first_parameter_resource(resource, function, *args, **kwargs):
+    resource_id = args[0]
+
+    return multiple_oneview.query_ov_client_by_resource(resource_id, resource,
+                                                        function, *args,
+                                                        **kwargs)
+
+
+def filter_uuid_parameter_resource(resource, function, *args, **kwargs):
+    if 'filter' not in kwargs:
+        return all_oneviews_resource(resource, function, *args, **kwargs)
+
+    filter_parameter = kwargs['filter']
+    resource_id = filter_parameter.split('=')[1]
+
+    return multiple_oneview.query_ov_client_by_resource(resource_id, resource,
+                                                        function, *args,
+                                                        **kwargs)
+
+
+def all_oneviews_resource(resource, function, *args, **kwargs):
+    all_results = multiple_oneview.search_resource_multiple_ov(resource,
+                                                               function, None,
+                                                               *args,
+                                                               **kwargs)
+
+    return all_results
+
+
+def spt_get_all_with_filter(resource, function, *args, **kwargs):
+    if 'filter' not in kwargs:
+        return all_oneviews_resource(resource, function, *args, **kwargs)
+
+    filter_parameter = kwargs['filter']
+
+    for filters in filter_parameter:
+        filter_data = filters.split('=')
+        if filter_data[0] == 'enclosureGroupUri':
+            resource_id = filter_data[1]
+            return \
+                multiple_oneview.query_ov_client_by_resource(resource_id,
+                                                             resource,
+                                                             function,
+                                                             *args, **kwargs)
+
+
+def create_server_profile(resource, function, *args, **kwargs):
+    sp = args[0]
+    spt_uuid = sp['description']
+    return _run_action(spt_uuid, 'server_profile_templates', 'get', resource,
+                       function, *args, **kwargs)
+
+
+def delete_server_profile(resource, function, *args, **kwargs):
+    sp_uuid = args[0]
+    return _run_action(sp_uuid, 'server_profiles', 'get', resource,
+                       function, *args, **kwargs)
+
+
+def update_power_state_server_hardware(resource, function, *args, **kwargs):
+    sh_uuid = args[1]
+    return _run_action(sh_uuid, 'server_hardware', 'get', resource,
+                       function, *args, **kwargs)
+
+
+def _run_action(resource_id, resource_get, function_get, resource,
+                function, *args, **kwargs):
+
+    # Mapping OneView for the resource_id
+    multiple_oneview.query_ov_client_by_resource(resource_id,
```

Comment:
Are we always executing a get before the run the original action (create action, for example)? 
 Even when the resource already cached? 
(Sorry if I thinking wrong)

Reply:
Nice point. I will change to just query if the resource is not already mapped.